modules/apps/portal-security/portal-security-content-security-policy/src/main/java/com/liferay/portal/security/content/security/policy/internal/servlet/filter/ContentSecurityPolicyFilter.java

```diff
@@ -141,4 +206,103 @@ private boolean _isExcludedURIPath(HttpServletRequest httpServletRequest) {
 	@Reference
 	private Portal _portal;
 
+	private static class ContentSecurityPolicyNonceIncludeServletResponse
+		extends HttpServletResponseWrapper {
```

Comment:
This looks like a reasonable response buffering implementation, but wondering if there are other examples of doing this in the code base that you followed?

Reply:
I checked all existing classes which extends HttpServletResponseWrapper but I didn't find any which seems to be suitable capturing the response content.

Reply:
Generally we try to avoid doing such buffering because there is a performance hit both in request+response cycles and also RAM on the server. That could be why it is hard to find existing code. 

Given that it is a new pattern and our priority right now is to start assessing impact of CSP on portal, I think we can go ahead with it. If it is merged, then I think we should consider adding a unit test for the buffering behaviour. But only after we starting receiving intel on that CSP impact also.